src/cart/crt_swim.c

```diff
@@ -1064,7 +1063,7 @@ static int64_t crt_swim_progress_cb(crt_context_t crt_ctx, int64_t timeout_us, v
 			uint64_t max_delay = swim_suspect_timeout_get() * 2 / 3;
 
 			if (delay > max_delay) {
-				D_ERROR("Network outage detected (idle during "
+				D_ERROR("SWIM network outage detected (idle during "
```

Comment:
I tend to think we should never update `csm->csm_last_unpack_hlc` in this code block, since that variable is the ground truth to measure how long the current node has been isolated. If we need to calculate progressive delay, let's introduce another variable for that.

```
if (csm->csm_alive_count > 2) {
```
Do you know what's the purpose of this check?

Reply:
I don't know about `csm_alive_count`.

Agreed on "should never update `csm_last_unpack_hlc`". But I wonder if we just remove the update below, would this error be printed too frequently? Perhaps this should be addressed in next PR that changes the handling of outages?

Reply:
It seems that `csm_alive_count` intends to track the number of alive members. (This tracking might have been broken by an old change of mine.) The number must be > 2 here perhaps because if there's only one alive member, that is, myself, then there's no use in detecting outage, whereas if there are two alive members, then we need to detect the death of the peer?